sdk/tests/deqp/functional/gles3/es3fBufferCopyTests.js

```diff
@@ -0,0 +1,363 @@
+            }
+        }
+
+        // .subrange
+
+        var subrangeGroup = new tcuTestCase.DeqpTest('subrange', 'Buffer subrange copy tests');
+        this.addChild(subrangeGroup);
+
+        /**
+         * @type {Array<{name: string, srcSize: number, dstSize: number, srcOffset: number, dstOffset: number, copySize: number}>}
+         */
+        var cases = [{
+                name: 'middle', srcSize: 1000, dstSize: 1000, srcOffset: 250, dstOffset: 250, copySize: 500
+            },{
+                name: 'small_to_large', srcSize: 100, dstSize: 1000, srcOffset: 0, dstOffset: 409, copySize: 100
+            },{
+                name: 'large_to_small', srcSize: 1000, dstSize: 100, srcOffset: 409, dstOffset: 0, copySize: 100
+            },{
+                name: 'low_to_high_1', srcSize: 1000, dstSize: 1000, srcOffset: 0, dstOffset: 500, copySize: 500
+            },{
+                name: 'low_to_high_2', srcSize: 997, dstSize: 1027, srcOffset: 0, dstOffset: 701, copySize: 111
+            },{
+                name: 'high_to_low_1', srcSize: 1000, dstSize: 1000, srcOffset: 500, dstOffset: 0, copySize: 500
+            },{
+                name: 'high_to_low_2', srcSize: 1027, dstSize: 997, srcOffset: 701, dstOffset: 17, copySize: 111
+        }];
+
+        for (var ndx = 0; ndx < cases.length; ndx++) {
+            var srcTarget = gl.COPY_READ_BUFFER;
+            var dstTarget = gl.COPY_WRITE_BUFFER;
+            var hint = gl.STATIC_DRAW;
+            verify = glsBufferTestUtil.VerifyType.AS_VERTEX_ARRAY;
+
+            subrangeGroup.addChild(
+                new es3fBufferCopyTests.BasicBufferCopyCase(
+                    cases[ndx].name, '',
+                    srcTarget, cases[ndx].srcSize, hint,
+                    dstTarget, cases[ndx].dstSize, hint,
+                    cases[ndx].srcOffset, cases[ndx].dstOffset, cases[ndx].copySize,
+                    verify
+                )
+            );
+        }
+
+        // .single_buffer
+
+        var singleBufGroup = new tcuTestCase.DeqpTest('single_buffer', 'Copies within single buffer');
+        this.addChild(singleBufGroup);
+
+        for (var srcTargetNdx = 0; srcTargetNdx < bufferTargets.length; srcTargetNdx++) {
+            for (var dstTargetNdx = 0; dstTargetNdx < bufferTargets.length; dstTargetNdx++) {
+                if (srcTargetNdx == dstTargetNdx)
+                    continue;
+
+                // In WebGL 2, we can't rebind an ELEMENT_ARRAY_BUFFER or TRANSFORM_FEEDBACK_BUFFER as a
+                // different type of buffer, so we skip those cases.
+                if (bufferTargets[srcTargetNdx] == gl.ELEMENT_ARRAY_BUFFER || bufferTargets[srcTargetNdx] == gl.TRANSFORM_FEEDBACK_BUFFER ||
+                    bufferTargets[dstTargetNdx] == gl.ELEMENT_ARRAY_BUFFER || bufferTargets[dstTargetNdx] == gl.TRANSFORM_FEEDBACK_BUFFER)
```

Comment:
Thanks for proactively modifying these tests to obey the WebGL 2.0 restrictions.